File: 563 - Binary Tree Tilt.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int absd = 0;
    public int findTilt(TreeNode root) {
        this.post(root);
        return this.absd;
    }
    public int post(TreeNode root) {
        if (root == null) {
            return 0;
        }
        int lsum = this.post(root.left);
        int rsum = this.post(root.right);
        this.absd += Math.abs(lsum - rsum);
        return root.val + lsum + rsum;
    }
}

File: 563 - Binary Tree Tilt.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
let absd;
var findTilt = function (root) {
    absd = 0;
    post(root);
    return absd;
};

function post(root) {
    if (root == null) {
        return 0;
    }
    let lsum = post(root.left);
    let rsum = post(root.right);
    absd += Math.abs(lsum - rsum);
    return root.val + lsum + rsum;
}
